agent_code/our_agent_v2/train.py

```diff
@@ -164,15 +171,29 @@ def game_events_occurred(self, old_game_state: dict, self_action: str, new_game_
     if not 'GOT_KILLED' in events:
         events.append(SURVIVED_STEP)         
     
-    ################## (2) Store Transision: #################
+    ################## (2) Compute n-step reward and store tuble in Transitions: #################
+
+    self.n_step_transitions.append((transform(self, old_game_state), self_action, transform(self, new_game_state), reward_from_events(self, events)))
     
-    self.transitions.append(Transition(transform(self, old_game_state), self_action, transform(self, new_game_state), reward_from_events(self, events)))
+    if len(self.n_step_transitions) == N_STEPS:
+        
+        n_step_reward = np.sum(((GAMMA)**np.arange(N_STEPS)).dot(np.array(self.n_step_transitions)[:,3]))
```

Comment:
This is what gives the errors I believe, calling np.array() on a deque storing tuples of different sizes and shapes.
Instead of forcing the entire list to an array, and since you only need the rewards, you could do
`
reward_arr = np.([self.n_step_transitions[i][-1] for i in range(N_STEPS)])
`
and then dot this with the gammas. And because you are dotting you don't need the np.sum, right?

Reply:
```suggestion
        reward_arr = np.array([self.n_step_transitions[i][-1] for i in range(N_STEPS)])
        n_step_reward = ((GAMMA)**np.arange(N_STEPS)).dot(reward_arr)
```